src/schema/v2/notifications/index.ts

```diff
@@ -98,6 +99,68 @@ export const NotificationType = new GraphQLObjectType<any, ResolverContext>({
         return formatDate(date, dateFormat, timezone)
       },
     },
+    previewImages: {
+      type: new GraphQLList(Image.type),
+      args: {
+        size: { type: GraphQLInt },
+      },
+      resolve: async (
+        notification,
+        { size },
+        { articleLoader, artworksLoader, showsLoader, viewingRoomLoader }
+      ) => {
+        switch (notification.activity_type) {
+          case "ViewingRoomPublishedActivity": {
+            const viewingRoomImages = await Promise.all(
+              notification.object_ids.map(async (viewingRoomId) => {
+                const { image_versions, image_url } = await viewingRoomLoader(
+                  viewingRoomId
+                )
+
+                return normalizeImageData({
+                  image_versions,
+                  image_url,
+                })
+              })
+            )
+
+            return compact(viewingRoomImages)
+          }
+          case "ArticleFeaturedArtistActivity": {
+            const images = await Promise.all(
```

Comment:
Using `articleLoader` in combination with `Promise.all` isn't ideal, but I wasn't able to easily make Positron's `articles` endpoint work with article IDs. I thought this might be an edge case anyway because most notifications should only have one article 😴

Reply:
Seems fine, it can be an optimization if there's a list endpoint that takes a list of id's, vs. hitting individual loaders (same for viewing rooms).